src/order/well_founded_set.lean

```diff
@@ -4,6 +4,7 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Aaron Anderson
 -/
 import algebra.pointwise
+import group_theory.submonoid.membership
```

Comment:
This import looks *very* wrong. It should definitely go the other way around. Would you mind moving ` submonoid_closure` to `group_theory.submonoid.membership`? Similarly, all the `pointwise` section belongs in `algebra.pointwise`.